### PR TITLE
httpapi: support /status to show PD version

### PR DESCRIPTION
**Pull request by choleraehyq**
See https://github.com/pingcap/pd/issues/551
PTAL @siddontang @nolouch 
Signed-off-by: Cholerae Hu <choleraehyq@gmail.com>

### CODE REVIEW

**Comment by disksing** [comment]
Please sign the CLA @choleraehyq .

File: server/api/status.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package api
```
Comment (disksing) [inline server/api/status.go]: Add license header at the top of the file.

File: server/api/status.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func newStatusHandler(rd *render.Render) *statusHandler {
+	return &statusHandler{
+		rd: rd,
+	}
+}
+
+func (h *statusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
```
Comment (siddontang) [inline server/api/status.go]: seem the default type is json, no need to set this, please ensure it.

Comment (choleraehyq) [inline server/api/status.go]: Yes. I removed it.

**Comment by nolouch** [comment]
LGTM

**Comment by siddontang** [comment]
Please add test

**Comment by choleraehyq** [comment]
@siddontang Test added.

File: server/api/status_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	for _, num := range numbers {
+		cfgs, _, clean := mustNewCluster(c, num)
+		defer clean()
+
+		parts := []string{cfgs[rand.Intn(len(cfgs))].ClientUrls, apiPrefix, "/api/v1/status"}
+		addr := mustUnixAddrToHTTPAddr(c, strings.Join(parts, ""))
+		resp, err := s.hc.Get(addr)
+		c.Assert(err, IsNil)
+		buf, err := ioutil.ReadAll(resp.Body)
+		c.Assert(err, IsNil)
```
Comment (siddontang) [inline server/api/status_test.go]: miss resp.Body.Close

File: server/api/status_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	c.Assert(got.BuildTS, Equals, server.PDBuildTS)
+	c.Assert(got.GitHash, Equals, server.PDGitHash)
+}
+
+func (s *testStatusAPISuite) TestStatus(c *C) {
+	numbers := []int{1, 3}
+
+	for _, num := range numbers {
+		cfgs, _, clean := mustNewCluster(c, num)
+		defer clean()
```
Comment (siddontang) [inline server/api/status_test.go]: don't call defer in for loop.

File: server/api/status_test.go (hunk 3 above)
Comment (choleraehyq) [inline server/api/status_test.go]: Done.

**Comment by siddontang** [comment]
PTAL @disksing

**Comment by choleraehyq** [comment]
@disksing Can this be merged?

**Pull request merged**